r_d11/r_d11.cpp

```diff
@@ -0,0 +1,133 @@
+#include <iostream>
+
+#define ROWS 3
+#define COLS 3
+
+bool find(int arr_2d[][COLS], int toFind, unsigned short rows = ROWS) {
+    if (rows <= 0 && COLS <= 0) return false;
+
+    for (unsigned short i = 0; i < rows; i++)
+    {
+        for (unsigned short j = 0;  j < COLS;  j++)
+        {
+            if (arr_2d[i][j] == toFind) return true;
+        }
+    }
+
+    return false;
+}
+
+enum class SortingDirection
+{
+    ASCENDING,
+    DESCENDING,
+};
+
+bool isSorted(const int* arr, int size, SortingDirection direction) {
+    if (size <= 0) return false;
+
+    switch (direction)
+    {
+    case SortingDirection::ASCENDING: {
+        int min = INT16_MIN;
+        for (unsigned int i = 0; i < size; i++)
+        {
+            if (min >= arr[i]) return false;
+            min = arr[i];
+        }
+        return true;
+    }
+    case SortingDirection::DESCENDING: {
+        int max = INT16_MAX;
+        for (unsigned int i = 0; i < size; i++)
+        {
+            if (max <= arr[i]) return false;
+            max = arr[i];
+        }
+        return true;
+    }
+    }
+}
+
+
+int stackColsRevers(int arr_2d[][COLS], int *&pointer, unsigned short rows = ROWS) {
+    pointer +=rows;
+    if (pointer - arr_2d[0] > ROWS * COLS) pointer -= (ROWS * COLS) + 1;
+    return *pointer;
+}
+
+int snakeRowsRevers(int arr_2d[][COLS], int*& pointer, unsigned short rows = ROWS) {
+    short int orientation = 1;
+    unsigned int distance = pointer - arr_2d[0];
+
+    if ((distance / (ROWS)) % 2) orientation = -1;
+    pointer += orientation;
+
+    distance = pointer - arr_2d[0];
+    if (orientation == 1 && ((distance / (ROWS)) % 2)) pointer -= (ROWS + 1);
+    if (orientation == -1 && !(distance / (ROWS)) % 2) pointer -= (ROWS - 1);
+
+    return *pointer;
+}
```

Comment:
That's is some crazy overcomplication for simple array traversal with 2 loops🥲  I don't think I have enough power of will to demystify that, I hope it does work)